app/src/main/java/live/hms/app2/ui/meeting/commons/VideoGridBaseFragment.kt

```diff
@@ -144,15 +143,15 @@ abstract class VideoGridBaseFragment : Fragment() {
       iconScreenShare.visibility = if (item.isScreen) View.VISIBLE else View.GONE
       iconAudioOff.visibility = visibility(
         item.isScreen.not() &&
-            (item.audio == null || item.audio!!.isMute)
+                (item.audio == null || item.audio!!.isMute)
       )
-      icDegraded.visibility = if(item.video?.isDegraded == true) View.VISIBLE else View.GONE
+      icDegraded.visibility = if (item.video?.isDegraded == true) View.VISIBLE else View.GONE
 
-      /** [View.setVisibility] */
-      val surfaceViewVisibility = if (item.peer.videoTrack == null
-        || item.video == null
-        || item.video?.isMute == true
-        || item.video?.isDegraded == true) {
+      val surfaceViewVisibility = if (
+        item.video == null ||
```

Comment:
We need `item.peer.videoTrack == null` check as well, because there may be cases when item.video is not null but the track has been removed from peer. If we dont put this check we may see black tiles in role change of a remote peer

Reply:
this was changed.